plugin/src/main/cpp/extensions/openxr_fb_spatial_entity_query_extension_wrapper.cpp

```diff
@@ -146,14 +146,13 @@ void OpenXRFbSpatialEntityQueryExtensionWrapper::on_space_query_results(const Xr
 
 	QueryInfo *query = queries.getptr(event->requestId);
 
-	Vector<XrSpaceQueryResultFB> results;
 	query->results.resize(queryResults.resultCountOutput);
-	queryResults.resultCapacityInput = query->results.size();
-	queryResults.resultCountOutput = 0;
+	queryResults.resultCapacityInput = queryResults.resultCountOutput;
```

Comment:
Should we check the value of `resultCountOutput` here and return early if it's `0` as done in https://github.com/GodotVR/godot_openxr_vendors/pull/341?

Reply:
Hm, logically, that makes sense. However, the spec has this:

> This function must only retrieve each query result once. After the application has used this function to retrieve a query result, the runtime frees its copy.

If we return early and don't do the second call to attempt to get the result data, will the runtime ever free this query?

If the query id wasn't valid, we'd be getting `XR_ERROR_VALIDATION_FAILURE` on this 2nd call, and we don't get that error if the query returns zero results. So, that leads me to believe that the runtime is keeping the query data alive until we do the 2nd call.

Here's the relevant part of the spec: https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#xrRetrieveSpaceQueryResultsFB